admin_terraform_version.go

```diff
@@ -70,29 +79,31 @@ type AdminTerraformVersionsListOptions struct {
-	Type             string  `jsonapi:"primary,terraform-versions"`
-	Version          *string `jsonapi:"attr,version"` // Required
-	URL              *string `jsonapi:"attr,url"`     // Required
-	Sha              *string `jsonapi:"attr,sha"`     // Required
-	Official         *bool   `jsonapi:"attr,official,omitempty"`
-	Deprecated       *bool   `jsonapi:"attr,deprecated,omitempty"`
-	DeprecatedReason *string `jsonapi:"attr,deprecated-reason,omitempty"`
-	Enabled          *bool   `jsonapi:"attr,enabled,omitempty"`
-	Beta             *bool   `jsonapi:"attr,beta,omitempty"`
+	Type             string                     `jsonapi:"primary,terraform-versions"`
+	Version          *string                    `jsonapi:"attr,version"` // Required
+	URL              *string                    `jsonapi:"attr,url"`     // Required
+	Sha              *string                    `jsonapi:"attr,sha"`     // Required
+	Official         *bool                      `jsonapi:"attr,official,omitempty"`
+	Deprecated       *bool                      `jsonapi:"attr,deprecated,omitempty"`
+	DeprecatedReason *string                    `jsonapi:"attr,deprecated-reason,omitempty"`
+	Enabled          *bool                      `jsonapi:"attr,enabled,omitempty"`
+	Beta             *bool                      `jsonapi:"attr,beta,omitempty"`
+	Archs            []*ToolVersionArchitecture `jsonapi:"attr,archs"`
 }
 
 // AdminTerraformVersionUpdateOptions for updating terraform version.
 // https://developer.hashicorp.com/terraform/enterprise/api-docs/admin/terraform-versions#request-body
 type AdminTerraformVersionUpdateOptions struct {
-	Type             string  `jsonapi:"primary,terraform-versions"`
-	Version          *string `jsonapi:"attr,version,omitempty"`
-	URL              *string `jsonapi:"attr,url,omitempty"`
-	Sha              *string `jsonapi:"attr,sha,omitempty"`
-	Official         *bool   `jsonapi:"attr,official,omitempty"`
-	Deprecated       *bool   `jsonapi:"attr,deprecated,omitempty"`
-	DeprecatedReason *string `jsonapi:"attr,deprecated-reason,omitempty"`
-	Enabled          *bool   `jsonapi:"attr,enabled,omitempty"`
-	Beta             *bool   `jsonapi:"attr,beta,omitempty"`
+	Type             string                     `jsonapi:"primary,terraform-versions"`
+	Version          *string                    `jsonapi:"attr,version,omitempty"`
+	URL              *string                    `jsonapi:"attr,url,omitempty"`
+	Sha              *string                    `jsonapi:"attr,sha,omitempty"`
+	Official         *bool                      `jsonapi:"attr,official,omitempty"`
+	Deprecated       *bool                      `jsonapi:"attr,deprecated,omitempty"`
+	DeprecatedReason *string                    `jsonapi:"attr,deprecated-reason,omitempty"`
+	Enabled          *bool                      `jsonapi:"attr,enabled,omitempty"`
+	Beta             *bool                      `jsonapi:"attr,beta,omitempty"`
+	Archs            []*ToolVersionArchitecture `jsonapi:"attr,archs"`
```

Comment:
Just clarifying on the omission of `omitempty` in the struct tag: is this field required in the request body? If not, we will end up serializing an empty array potentially overriding any existing architectures persisted in the API. Normally for optional fields, we use `omitempty` to prevent this from occurring. 

Furthermore, if we do want to allow deleting all architectures for a given TF version then we will need to define a separate method that intentionally allows an empty slice to be sent over the wire, i.e does not include `omitempty`. See #1023 as an example.

Reply:
Turns out I don't need to edit the `AdminTerraformVersionUpdateOptions` struct for what I am trying to accomplish. I have added `omitempty` to the `Archs` field on the `AdminTerraformVersionCreateOptions` struct. Thanks for catching this.